source/_integrations/renault.markdown

```diff
@@ -50,10 +50,17 @@ Cancel A/C on vehicle.
 
 Update charge schedule on vehicle.
 
-  | Service data attribute | Required | Description |
-  | ---------------------- | -------- | ----------- |
+  | Service data attribute | Required | Description | Example |
+  | ---------------------- | -------- | ----------- | ------- |
   | `vehicle`| yes | device_id of the vehicle |
-  | `schedules` | yes | Schedule details. Can be a single schedule or a list of schedules |
+  | `schedules` | yes | Schedule details. Can be a single schedule or a list of schedules | `[{'id':1,'activated':true,'monday':{'startTime':'T12:00Z','duration':15}},<br>{'id':2,'activated':false,'monday':{'startTime':'T12:00Z','duration':240}}<br>]` |
+  
+Notes:
+
+- `schedules` can be in the form `{'id':1,...}` when updating a single schedules, or in the form `[{'id':1,...},{'id':2,...},...]` when updating multiple schedules within the same call
+- the `id` is compulsory on each `schedule` (should be 1 to 5 depending on the vehicle)
+- the `activated` flag is an optional boolean. If it is not provided, then the existing flag will be kept as is.
+- the `monday` to `sunday` elements are optional. If they are not provided, then the existing settings will be kept for each day. If they are provided as None, then the existing setting will be cleared. If a value is provided, it must conform to this format `{'startTime':'T12:00Z','duration':15}` where start time is in UTC format and the duration is in minutes.
```

Comment:
I noticed now that we only validate the time string as string in the service schema. Could we use the time validator to help users use the correct time string?

https://github.com/home-assistant/core/blob/33fb080c1e7dc5db8e24cc142fd9b1972ef337f3/homeassistant/helpers/config_validation.py#L351-L364

Then we can format the final string from the returned datetime.

Reply:
The CLI version of RenaultAPI has some validation/conversion built in: https://github.com/hacf-fr/renault-api/blob/7c2336c52ed0298a3ed2b8e7dbb77b839676dced/src/renault_api/cli/charge/schedule.py#L202

There is also a draft PR to add validation to the non-CLI version: https://github.com/hacf-fr/renault-api/pull/157

There are some extra rules like the minutes of startTime needs to be in (00,15,30,45) and the total duration needs to be a multiple of 15 minutes (eg. 240 is valid but not 245)
There are also some overlap rules (Monday 23:45 for 120 minutes and Tuesday 00:15 for 30 minutes is invalid)

I'll have a go at a follow-up PR.

Reply:
I'm not sure when I'll get a chance to work on the new PR.
Should this documentation PR be merged anyway in the meantime?

Reply:
Sure! The docs are fine.